core/src/main/java/com/linecorp/armeria/common/AbstractContextAwareExecutorService.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+abstract class AbstractContextAwareExecutorService implements ExecutorService {
+    final ExecutorService executor;
+
+    AbstractContextAwareExecutorService(ExecutorService executor) {
+        this.executor = executor;
+    }
+
+    @Nullable
+    protected abstract RequestContext context();
```

Comment:
Could remove `protected`
```suggestion
    abstract RequestContext context();
```